src/reconcile/rec_visibility.c

```diff
@@ -342,7 +350,7 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
                 upd_select->durable_ts = upd_select->start_ts = upd->start_ts;
             if (upd->txnid < upd_select->stop_txn)
                 upd_select->start_txn = upd->txnid;
-        } else {
+        } else if (!__wt_txn_upd_visible_all(session, orig_selected)) {
```

Comment:
I don't understand this part of the check. If the tombstone is not globally visible, we're going to leave the on-disk value untouched. Why is that a requirement for writing the stop time pair?

Doesn't this mean we're going to end up with a tombstone update in lookaside that is newer than the on-disk value?

Reply:
Oh, I thought we would delete the key if nothing is selected to write to disk. If we only leave it untouched, this this is wrong. I'll fix.